ember-core/src/main/scala/org/http4s/ember/core/EmptyStreamError.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.core
+
+private[ember] final case class EmptyStreamError() extends Exception("Cannot Parse Empty Stream")
```

Comment:
Nit: I don't love exceptions being case classes.  It implies equality where there is none.  I see you're using it as an extractor, but I usually catch these by type.

Reply:
This is also not a hill I'll die on, but every other exception type defined in that package is also a case class, and most of them are not used in a match clause :) . If we want to go one way or the other, I'm all for for following well-defined guidelines

Reply:
Agree, I care about consistency more than which way we go.  Let's merge, and we can revisit the case class bikeshed.